tests/parser-test-files/parse-expression.txt

```diff
@@ -0,0 +1,299 @@
+================================================================================
+Parse expression
+================================================================================
+
+(package coalton-unit-tests)
+
+(define f 5)
+
+================================================================================
+Parse expression
+================================================================================
+
+(package coalton-unit-tests)
+
+(define (f _x) 5)
+
+================================================================================
+Parse expression
+================================================================================
+
+(package coalton-unit-tests)
+
+(define (f _x _y) 5)
+
+================================================================================
+Parse expression
+================================================================================
+
+(package coalton-unit-tests/prelude
+  (import coalton-prelude))
+
+(define f
+  (fn (x)
+    (+ x x)))
+
+================================================================================
+Parse expression
+================================================================================
+
+(package coalton-unit-tests/prelude
+  (import coalton-prelude))
+
+(define f
+  (fn ()
+    (+ 1 2)))
+
+================================================================================
+Parse expression (:disable)
+================================================================================
+
+(package coalton-unit-tests/prelude
+  (import coalton-prelude))
+
+(define (f)
+  (let ((x 1)
+        (declare y (T -> T))
+        (y (+ 1 2)))
+    (f x y)))
+
+================================================================================
+Parse expression
+================================================================================
+
+(package coalton-unit-tests/prelude
+  (import coalton-prelude))
+
+(define (f a b c)
+  (lisp (Integer -> Integer) (a b c)
+    (cl:+ 1 2)))
+
+================================================================================
+Parse expression (:disable)
+================================================================================
+
+(package coalton-unit-tests/prelude
+  (import coalton-prelude))
+
+(define (f x)
+  (match x
```

Comment:
This is invalid for a number of reasons, the first being that the nested `Some` makes the next match expect a type `(Optional (Optional :a))`

This isn't as robust but might be a worthy replacement in lieu of more match tests:
```
(declare f ((Default :a) => (Optional :a) -> :a))
  (define (f x)
    (match x
      ((Some y) y)
      (_ (default))))
```